src/csharp.tmLanguage.yml

```diff
@@ -1597,7 +1593,7 @@ repository:
   when-clause:
     begin: (?<!\.)\b(when)\b\s*(\()
     beginCaptures:
-      '1': { name: keyword.control.when.cs }
+      '1': { name: keyword.control.conditional.when.cs }
```

Comment:
Perhaps `exception.when` is more appropriate here, since it technically belongs to the "the try statement".